server/controllers/article.controller.js

```diff
@@ -204,5 +210,52 @@ class ArticleController {
     await sendMail(mailOption);
     return Response(res, 200, 'Article shared successfully');
   }
+
+  /**
+   * @description Get top rated articles controller
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} Response
+   */
+  static async getTopArticle(req, res) {
+    const allarticles = await Article.findAll({
+      where: {
+        published: true,
+      },
+      attributes: {
+        exclude: ['published', 'userId']
+      },
+      include: [{
+        model: User,
+        as: 'author',
+        attributes: {
+          exclude: ['istokenreset', 'password', 'social', 'roleId', 'subscribed', 'id']
+        }
+      }]
+    });
+    const rate = allarticles.map(async (article) => {
+      const ratings = await getArticleRating(article.dataValues.id);
+      const {
+        id, slug, title, description, body, image, tagList, createdAt, updatedAt, author
+      } = article;
+      const rateObj = {
+        id,
+        slug,
+        title,
+        description,
+        body,
+        image,
+        tagList,
+        createdAt,
+        updatedAt,
+        author,
+        ratings
+      };
+      return rateObj;
+    });
+    const rateResponse = await Promise.all(rate);
+    const response = rateResponse.sort((a, b) => b.ratings - a.ratings);
+    return Response(res, 200, 'Retrived all top rated articles feed', { article: response });
```

Comment:
Hi @davidokonji Great work. Just fix the typo at `Retri(e)ved`

Reply:
Thanks for the observation, fixed